### PR TITLE
Refactors ResultSet class to use properties and remove finalise method

### LINKED ISSUE #159: ResultSet Private Methods
The generation methods called by finalise in the ResultSet class should be private and only ever called by finalise.

The docs will also need to reflect this change.

### PR DESCRIPTION
**Pull request by meatballs**
Fixes #159

### CODE REVIEW

**Comment by meatballs** [comment]
Hold off on this for a moment - I'm doing a little more work on it.

**Comment by meatballs** [comment]
Should be good to go now.

**Comment by meatballs** [comment]
Finished with it now. Should be good to go.

**Comment by meatballs** [comment]
I'm done with this. No more fiddling. Hands are firmly in pockets.

File: docs/usage.rst [hunk 1]
```diff
@@ -76,34 +76,31 @@ Now to run the tournament and save the results::
     results = tournament.play()
 
 The output of `tournament.play()` is a `ResultSet` which is something that holds all the information we could need.
-First, let us generate the scores::
+First, let us view the scores::
```
Comment (drvinceknight) [inline docs/usage.rst]: So this line here is definitely appearing in readthedocs for me.

**Pull request merged**